File: 0415-add-strings/0415-add-strings.php
class Solution {

    /**
     * @param String $num1
     * @param String $num2
     * @return String
     */
    function addStrings($num1, $num2) {
        $n1Len = strlen($num1) - 1;
        $n2Len = strlen($num2) - 1;
        $carry = 0;
        $ans = '';
        
        while($n1Len >= 0 || $n2Len >= 0 || $carry > 0){
            
            //substract '0' helps to convert string to int
            $n1Val = $n1Len >= 0 ? $num1[$n1Len] - '0' : 0;
            $n2Val = $n2Len >= 0 ? $num2[$n2Len] - '0' : 0;
            
            $tmpSum = $n1Val + $n2Val + $carry;            
            $carry = intdiv($tmpSum, 10);
            $ans = ($tmpSum % 10) . $ans;

            
            $n1Len--;
            $n2Len--;
        }
        
        return $ans;
    }
}

File: 0415-add-strings/0415-add-strings.go
func addStrings(num1 string, num2 string) string {
    
    i := len(num1) - 1
    j := len(num2) - 1
    carry := 0

    var ans string
    
    for i >= 0 || j >= 0 || carry > 0 {
        sum := carry
        
        if i >= 0 {
            sum += int(num1[i] - '0')
        }
        
        if j >= 0 {
            sum += int(num2[j] - '0')
        }
        
        carry = sum / 10
        ans = string(sum % 10 + '0') + ans
        
        i--
        j--
    }
    
    return ans
}